pkg/devicemapper/devmapper.go

```diff
@@ -67,12 +67,14 @@ var (
 	ErrBusy                 = errors.New("Device is Busy")
 	ErrDeviceIDExists       = errors.New("Device Id Exists")
 	ErrEnxio                = errors.New("No such device or address")
+	ErrEnoData              = errors.New("No data available")
 )
 
 var (
-	dmSawBusy  bool
-	dmSawExist bool
-	dmSawEnxio bool // No Such Device or Address
+	dmSawBusy    bool
+	dmSawExist   bool
+	dmSawEnxio   bool // No Such Device or Address
+	dmSawEnoData bool // No data available
```

Comment:
Could these be returned from `task.run()` instead of globals?

Reply:
@dnephin The value is passed through a log callback:
https://github.com/moby/moby/blob/master/pkg/devicemapper/devmapper_log.go#L42-L56

which is wrapped in cgo:
https://github.com/moby/moby/blob/master/pkg/devicemapper/devmapper_wrapper.go#L13

I think it might be difficult to return from `task.run()`

Reply:
ok, yes that does seem like it would require a much bigger change to fix.